hivemind/optim/optimizer.py

```diff
@@ -344,6 +344,13 @@ def _compute_schema_hash(self) -> int:
     def is_alive(self) -> bool:
         return self.state_averager.is_alive()
 
+    @property
+    def gathered_peers(self) -> int:
```

Comment:
@justheuristic Is it a good idea to expose this or this info is already exposed somewhere?

Also, I'd expect a property named `gathered_peers` to return a collection of peers (e.g., peer IDs), not just the count of them. An alternative is to rename this property to something like `num_gathered_peers`.

Reply:
Hi, @IAL32 !

I recognize that it is convenient to keep track of group size, but merging this specific code into hivemind may introduce confusion in some modes. If use_local_updates=False, optimizer will have two different averaging rounds (for state and for gradients), and it is not clear which of them we should report. Furthermore, if [moshpit](https://arxiv.org/abs/2103.03239) is enabled, optimizer can run multiple concurrent averaging rounds.

If you keep track of it in your experiments, you can subclass hivemind.Optimizer as such:

```python

class MyOptimizer(hivemind.Optimizer)
    @property
    def gathered_peers(self) -> int:
        return len(self.state_averager.gathered_peers)

    def _make_state_averager(self, **kwargs) -> TrainingStateAverager:
        # https://github.com/learning-at-home/hivemind/blob/master/hivemind/optim/optimizer.py#L282-L297
        return MyModifiedStateAverager(see_base_funciton_code(**kwargs))

class MyModifiedStateAverager(TrainingStateAverager):
    gathered_peers = ...

```